### ISSUE TITLE
Training GMM-HMM results in: [RuntimeWarning: divide by zero encountered in log] 

### ISSUE THREAD

**Issue by Schiboni** [jmschrei/pomegranate#371]
Hello,
I am training GMM-HMM  by creating a specific model firstly, i.e., left-to-right, and then running .fit function. I am using multiprocessing for python on clusters of a server. I am running several instances of the traning to pick the best result.
I get this error at some point:

Traceback (most recent call last):
  File "/usr/lib/python3.4/threading.py", line 920, in _bootstrap_inner self.run()
  File "/usr/lib/python3.4/threading.py", line 868, in run self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.4/multiprocessing/pool.py", line 420, in _handle_results task = get()
  File "/usr/lib/python3.4/multiprocessing/connection.py", line 251, in recv
    return ForkingPickler.loads(buf.getbuffer())
  File "pomegranate/gmm.pyx", line 99, in pomegranate.gmm.GeneralMixtureModel.__init__
  File "pomegranate/bayes.pyx", line 89, in pomegranate.bayes.BayesModel.__init__
RuntimeWarning: divide by zero encountered in log

I do not understand exactly why but the code calculates the log of the weights. at line 89 of Bayesmodel.__init__. How can I solve the problem?
Thanks in advance

**Comment by jmschrei** [comment]
The log of the weights is calculated for numerical stability. Is it possible for you to provide a minimally reproducing example? That would allow me to debug the issue the fastest.

**Comment by Schiboni** [comment]
Thanks for you ready answer! 
To provide a minimally reproducing example is not that easy. Two main problems i have with that are:
1) the train data i am using, i.e., inertial sensor data of gestures, occupies a lot of space.
2) the problem occurs at some point after several random initialisation. Some time can take up to 2 hours before it shows up.
If you could give me a hint a could treat such cases as exceptions. Do the warning occur when a mixture's weight is 0? If the weight is 0, would that be a bug or would that depend on data themselves? Hope i am asking pertinent questions.

**Comment by jmschrei** [comment]
The error is being raised because one component in the mixture has 0 weight, meaning that a sample can never belong to it. This likely happened because a component began modeling some anomaly, and suddenly had too little data to support it. You might be making your model too complex. A simplification is essentially to imagine that you have k-means with some large number of centroids, and eventually one gets pushed off to model a single outlier data point. It's not an "error" persay, as much as it is just a bad initialization. If your models that are trained and don't encounter this error seem reasonable, I think that you can safely ignore these models as being an exception. 

tl;dr it ~looks~ like it's an issue with the data more than the code, because having a GMM where one component is impossible is not great. It should probably raise a readable error versus a divide by zero error though.

**Comment by jmschrei** [comment]
I'm going to mark this as resolved. Please re-open or open a new issue if you encounter problems.
